File: content/solid_template/scripts/vscripts/addon_game_mode_client.ts
/** @noSelfInFile **/

import { reloadable } from "./lib/tstl-utils";

SendToConsole("dota_combine_models 0");
Convars.SetBool("dota_combine_models", false);

require("requires");

@reloadable
class MClient extends CModule {
	initPriority(): number {
		return 1;
	}
	init(bReload: boolean) {
		if (!bReload) {
		}
		GameEvent("cl_script_reload", () => {
			require("addon_game_mode_client");
		}, undefined);
	}
}
declare global {
	/**
	 * @ClientOnly
	 */
	var Client: MClient;
}

Client ??= new MClient();

if (!GameModeActivated) {
	CModule.initialize();
}

require("reload");

File: game/solid_template/scripts/vscripts/addon_game_mode_client.lua
local ____lualib = require("lualib_bundle")
local __TS__SourceMapTraceBack = ____lualib.__TS__SourceMapTraceBack
__TS__SourceMapTraceBack(debug.getinfo(1).short_src, {["4"] = 3,["5"] = 4,["6"] = 6,["7"] = 8,["8"] = 9});
SendToConsole("dota_combine_models 0")
Convars:SetBool("dota_combine_models", false)
require("requires")
if not GameModeActivated then
    CModule:initialize()
end
